apps/src/templates/teacherDashboard/shapes.jsx

```diff
@@ -0,0 +1,25 @@
+import { PropTypes } from 'react';
+
+export const sectionShape = PropTypes.shape({
+  id: PropTypes.number.isRequired,
+  name: PropTypes.string.isRequired,
+  // Though we validate valid login types here, the server actually owns the
+  // canonical list, and passes us the list of valid login types.
+  loginType: PropTypes.oneOf(['word', 'email', 'picture']).isRequired,
```

Comment:
Yes. And valid_grades most likely. One of the things I'd like to clean up, but not the first thing on my list.